packages/@aws-cdk/aws-codepipeline-api/lib/action.ts

```diff
@@ -192,13 +211,19 @@ export abstract class Action extends cdk.Construct {
     }
   }
 
-  protected addOutputArtifact(name: string): Artifact {
-    const artifact = new Artifact(this, name);
+  protected addOutputArtifact(name?: string): Artifact {
+    const outputArtifactName = name === undefined
+      ? this.stage._generateOutputArtifactName(this)
+      : name;
+
+    const artifact = new Artifact(this, outputArtifactName);
     return artifact;
   }
 
-  protected addInputArtifact(artifact: Artifact): Action {
-    this._inputArtifacts.push(artifact);
+  protected addInputArtifact(artifact?: Artifact): Action {
```

Comment:
Similar sugaring tip as before applies there.

Reply:
Same reply as above (I'm actually using the optional parameter).